metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -329,6 +330,23 @@ def task_pre_step(
                 "METAFLOW_KUBERNETES_SERVICE_ACCOUNT_NAME"
             ]
             meta["kubernetes-node-ip"] = os.environ["METAFLOW_KUBERNETES_NODE_IP"]
+
+            # Capture AWS instance identity metadata. This is best-effort only since
+            # access to this end-point might be blocked on AWS and not available
+            # for non-AWS deployments.
+            # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+            # TODO (savin): Introduce equivalent support for Microsoft Azure and
+            #               Google Cloud Platform
+            try:
+                instance_meta = requests.get(
+                    url="http://169.254.169.254/latest/dynamic/instance-identity/document"
```

Comment:
Is this IP significant?

Reply:
Yes, it's a magic IP that hosts instance metadata for AWS, Azure and GCP